pyrenew/latent/infections.py

```diff
@@ -80,14 +80,21 @@ def sample(
         InfectionsSample
             Named tuple with "infections".
         """
-        if I0.size < gen_int.size:
+        if I0.shape[0] < gen_int.size:
             raise ValueError(
                 "Initial infections vector must be at least as long as "
                 "the generation interval. "
-                f"Initial infections vector length: {I0.size}, "
+                f"Initial infections vector length: {I0.shape[0]}, "
                 f"generation interval length: {gen_int.size}."
             )
 
+        if I0.ndim != Rt.ndim:
+            raise ValueError(
+                "Initial infections and Rt must have the same dimensions. "
+                f"Got initial infections of shape {I0.shape} "
+                f"and Rt of shape {Rt.shape}."
+            )
```

Comment:
Error message (about shapes) and check (about dimensional count) are slightly inconsistent.

But is this really the check we want? One could maybe enforce `I0.shape[1:] == Rt.shape[1:]`, but with `I0.shape[0]` and `Rt.shape[0]` potentially different. 

Even that might be too strong: for example, one might imagine applying a single `Rt` timeseries of shape `(20,)` with an `I0` of shape `(10, 5)` (5 different initial timeseries, each of length 10 timesteps).

Reply:
The checks on the shapes of `I0` and `Rt` is not required in `infections.py` but helps ensure that `Rt` and `R_adjustment` are compatible for broadcasting in `infectionswithfeedback.py`.

I agree enforcing equal batch dimension with `I0.shape[1:] == Rt.shape[1:]` is a better check.

Reply:
Sounds good. LMK when ready for re-review.